### PR TITLE
Combine individuals and sample set tables

### LINKED ISSUE #16: Individuals: Colours on map not in table
The map shows sample set colours, but these colours aren’t reflected in the table. Users have to switch between pages or remember the colours. Table should show colours matching the map.

### LINKED ISSUE #18: Indivuduals: Sample set names not in individuals table
The table doesn’t show the sample set names. It’s hard to identify which set is being referred to. Sample set names should be added to the table. Users have to switch between pages or remember the what samples have which ids.

### LINKED ISSUE #114: Individuals: Adding new sample set does not automatically update the sidebar settings
Adding a new sample set does not automatically update the options of the sidebar settings. Currently the page has to be updated before these are available as options.

image
image

I added a comment in the tooltip that one should manually update the page after adding new sample sets, as I could not get it to automatically update.

### PR DESCRIPTION
**Pull request by almanilsson**
This PR combines the two tables (thus finally closes #16 and closes #18) and adds a global refresh button as an improvement of the temporary fix which closes #114.

image

### CODE REVIEW

File: src/tseda/datastore.py [hunk 1]
```diff
@@ -252,14 +252,17 @@ class IndividualsTable(Viewer):
     sample_sets_table = param.ClassSelector(class_=SampleSetsTable)
 
     columns = [
-        "name",
+        "color",
         "population",
         "sample_set_id",
-        "selected",
+        "name_sample",
```
Comment (percyfal) [inline src/tseda/datastore.py:258]: The information in this column refers to the population, so maybe call the variable `name_population`?

Comment (percyfal) [reply src/tseda/datastore.py:258]: Or maybe `name_sample_set` (`sample_set_name`)?

File: src/tseda/datastore.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ class SampleSetsTable(Viewer):
     )
 
     create_sample_set_warning = pn.pane.Alert(
-        "If the new sample set does not immediately show, click Update below",
+        "If the new sample set is not shown immediately, click Update above",
```
Comment (percyfal) [inline src/tseda/datastore.py:87]: The button says "Refresh"

File: src/tseda/datastore.py [hunk 3]
```diff
@@ -326,6 +336,11 @@ class IndividualsTable(Viewer):
             "tristate": True,
             "indeterminateValue": None,
         },
+        "name_sample": {
```
Comment (percyfal) [inline src/tseda/datastore.py:339]: See comment on naming above.

Comment (percyfal) [inline src/tseda/datastore.py:339]: See comment on naming above.

File: src/tseda/datastore.py [hunk 4]
```diff
@@ -423,10 +443,47 @@ def check_data_modification(self):
...
 
+    def combine_tables(self, individuals_table):
+        """Combine individuals and sample sets table."""
+
+        combined_df = pd.merge(
+            individuals_table.rx.value,
+            self.sample_sets_table.data.rx.value,
+            left_on="sample_set_id",
+            right_index=True,
+            suffixes=("_indiv", "_sample"),
```
Comment (percyfal) [inline src/tseda/datastore.py:454]: Again, I would suggest using `_sample_set`, otherwise I think of a single sample (like an individual)

Comment (almanilsson) [reply src/tseda/datastore.py:454]: Agreed! I will change that but will keep it as suffixes as pandas merge does not give the option to use prefixes

**Comment by almanilsson** [comment]
> If you change Update to Refresh (or the other way round) you can go ahead and merge.

Right. Thank you for the reminder

**Pull request merged**